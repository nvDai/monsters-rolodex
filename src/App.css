.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.title {
  font-family: 'Bigelow Rules', cursive;
  font-size: 72px;
  color: #0ccac4;
}
