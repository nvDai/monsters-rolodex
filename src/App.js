import React, { Component } from "react";
import "./App.css";
import CardList from './components/card-list/card-list.component'
import SearchBox from './components/search-box/search-box.component'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      monsters: [],
      searchFeild: ""
    };
    this.showProps = props
  }

  async componentDidMount() {
    const monsters = await this.fetchData()
    this.setState({ monsters })
  }

  fetchData = async () => {
    try {
      const res = await fetch("https://jsonplaceholder.typicode.com/users")
      const monsters = await res.json()
      return monsters
    } catch (error) {
      console.error(error)
    }
  }

  filteredMonsters = () => {
    const { monsters, searchFeild } = this.state;
    
    return monsters.filter(monster =>
      monster.name.toLowerCase().includes(searchFeild.toLowerCase())
    );
  }

  handleChangeInput = (e) => {
    this.setState({ searchFeild: e.target.value })
  }

  render() {
    return (
      <div className="App">
        <h1 className="title">Monster Rolodex</h1>
        <SearchBox
          placeholder="Search monster"
          handleChangeInput={this.handleChangeInput}
        />
        <CardList monsters={this.filteredMonsters()} />
      </div>
    );
  }
}

export default App;
