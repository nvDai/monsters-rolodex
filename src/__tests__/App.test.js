import React from 'react';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import nock from 'nock'

import App from '../App';

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  });

  it('should renders without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should filter monters correctly', () => {
    const filteredMonsters = [
      { 
        id: 2,
        name: 'Michael',
        email: 'mike@example.com'
      }
    ]

    wrapper.setState({
      monsters: [
        { 
          id: 2,
          name: 'Michael',
          email: 'mike@example.com'
        }
      ],
      searchFeild: 'Michael'
    })

    expect(wrapper.instance().filteredMonsters()).toEqual(filteredMonsters)
  })

  it('should not filter any monters', () => {
    wrapper.setState({
      monsters: [
        { 
          id: 2,
          name: 'Michael',
          email: 'mike@example.com'
        }
      ],
      searchFeild: 'mike'
    })

    expect(wrapper.instance().filteredMonsters()).toEqual([])
  })

  it('should be fetching data successfull with nock API', () => {
    const mockMonsters = [
      {
        id: '1',
        name: 'Michael',
        email: 'michael@gmail.com'
      },
      {
        id: '2',
        name: 'Michael 2',
        email: 'michael2@gmail.com'
      }
    ]
    const nockAPI = nock('https://randommonsters.me')
    .get('/monsters/')
    .reply(200, {
      results: mockMonsters,
    }, { 
      'Access-Control-Allow-Origin': '*', 
      'Content-type': 'application/json' 
    })
    
    return fetch("https://randommonsters.me/monsters/")
    .then(res => res.json())
    .then(data => {
      wrapper.setState({monsters: data.results})
      expect(wrapper.instance().state['monsters']).toEqual(mockMonsters)
      nockAPI.done()
    })
  })

  it('should be fetching data failure', () => {
    nock('https://randommonsters.me')
    .get('/monsters/')
    .reply(500, { results: { error: 'Fetching data failure' }}, { 
      'Access-Control-Allow-Origin': '*', 
      'Content-type': 'application/json' 
    })
    
    return fetch("https://randommonsters.me/monsters/")
    .then(res => res.json())
    .then(res => {
      expect(res.results.error).toBe('Fetching data failure')
    })
    .catch(err => console.log(err))
  })
})

