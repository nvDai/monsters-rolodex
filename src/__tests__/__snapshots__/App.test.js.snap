// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders without crashing 1`] = `
<div
  className="App"
>
  <h1
    className="title"
  >
    Monster Rolodex
  </h1>
  <SearchBox
    handleChangeInput={[Function]}
    placeholder="Search monster"
  />
  <CardList
    monsters={Array []}
  />
</div>
`;
